contracts/truefi2/LoanFactory2.sol

```diff
@@ -34,7 +35,7 @@ contract LoanFactory2 is ILoanFactory2, Initializable {
     // ========= IN STORAGE CORRUPTION ===========
 
     // @dev Track Valid LoanTokens
-    mapping(IDebtToken => bool) public override isLoanToken;
+    mapping(ILoanToken2 => bool) public override isLoanToken;
```

Comment:
This should become ILegacyLoanToken2.

At the bottom we should also add a mapping for ILoanToken3s created by FTLA.

Reply:
Why should it be legacy loan token? Don't see a reason why we'd want to keep old loans and new ones in different mappings.

Reply:
Separating mappings when we break backwards compatibility provides three benefits for us:
1. Type safety within our repo. We wouldn't need to typecast loans into TL2 loans and FTLA loans when we interact with a single mapping that's tracking both. This would help our compiler catch type errors that we might overlook. (Typecasting and untyping addresses are code smells that tells us we're fighting the compiler instead of letting it help us check our work.)
2. Easier on-chain monitoring of a deprecation/migration. It's much easier to check that a mapping or array is empty than to loop through all addresses, check ABIs to see whether they match the old or new version, and reduce to determine whether we're still using the old version anywhere.
3. Segregation of old and new handling logic. Breaking backwards compatibility is just creating new functions while leaving old code untouched. Deprecation is just deleting old code while leaving new code untouched.

Granted, _this_ particular mapping doesn't benefit much from points 1-3, but if we're segregating logic for TL2 loans and FTLA loans elsewhere, then point 4 below applies:
4. Consistency across a repo.

I'd be happy to keep this in a single mapping if we have a compelling case for it. One possibility I see is that `isLoanToken()` gets called in several places that we'd have to split into ugly case logic for TL2 and FTLA. But my hope is that the places where we call `isLoanToken()` vs. `isLegacyLoanToken()` (or whatever we'd like for TL2) are exactly the places we plan to split LT2 and FTLA implementations into separate functions.